theano/gpuarray/dnn_base.c

```diff
@@ -42,7 +42,8 @@ c_set_tensorNd(PyGpuArrayObject *var, cudnnTensorDescriptor_t desc) {
     strs[i] = 1;
     dims[i] = 1;
   }
-
+  //only for grouped convolution i.e when groups > 1
+  dims[1] = dims[1] / groups;
```

Comment:
I mean that you use `dims[1] =  dims[1] / groups` here and `dims[1] /= groups` in the other.